### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by LBHMGeorgieva**
What has been done?
- Renamed files
- Basic controller implementation + tests
- Use case implementation for calling jwt generate helper + basic test coverage
- Jwt generate helper implementation
- Boundaries created

### CODE REVIEW

File: ApiAuthTokenGenerator.Tests/V1/E2ETests/ExampleTest.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-namespace BaseApi.Tests.V1.E2ETests
+namespace ApiAuthTokenGenerator.Tests.V1.E2ETests
 {
     //For guidance on writing integration tests see the wiki page https://github.com/LBHackney-IT/lbh-base-api/wiki/Integration-Tests
     public class ExampleTest : IntegrationTests<Startup>
```
Comment (humulla) [inline ApiAuthTokenGenerator.Tests/V1/E2ETests/ExampleTest.cs:4]: Do we not want to change this file name to be linked the API?

File: ApiAuthTokenGenerator.Tests/V1/Gateways/ExampleGatewayTests.cs [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-using BaseApi.V1.Domain;
-using BaseApi.V1.Gateways;
+using ApiAuthTokenGenerator.Tests.V1.Helper;
+using ApiAuthTokenGenerator.V1.Domain;
+using ApiAuthTokenGenerator.V1.Gateways;
 using FluentAssertions;
 using NUnit.Framework;
 
-namespace BaseApi.Tests.V1.Gateways
+namespace ApiAuthTokenGenerator.Tests.V1.Gateways
```
Comment (humulla) [inline ApiAuthTokenGenerator.Tests/V1/Gateways/ExampleGatewayTests.cs:8]: Do we not want to change the file name to link to this one

File: ApiAuthTokenGenerator.Tests/V1/Infrastructure/ExampleContextTests.cs [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
-using BaseApi.V1.Infrastructure;
+using ApiAuthTokenGenerator.Tests.V1.Helper;
+using ApiAuthTokenGenerator.V1.Infrastructure;
 using NUnit.Framework;
 
-namespace BaseApi.Tests.V1.Infrastructure
+namespace ApiAuthTokenGenerator.Tests.V1.Infrastructure
 {
     [TestFixture]
     public class DatabaseContextTest : DatabaseTests
```
Comment (humulla) [inline ApiAuthTokenGenerator.Tests/V1/Infrastructure/ExampleContextTests.cs:9]: File name needs to be changed

File: ApiAuthTokenGenerator/V1/Controllers/ApiAuthTokenGeneratorController.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+using ApiAuthTokenGenerator.V1.UseCase.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Net.Mime;
+
+namespace ApiAuthTokenGenerator.V1.Controllers
+{
+    [ApiController]
+    //TODO: Rename to match the APIs endpoint
```
Comment (humulla) [inline ApiAuthTokenGenerator/V1/Controllers/ApiAuthTokenGeneratorController.cs:12]: Comment no longer required

File: ApiAuthTokenGenerator/V1/Controllers/ApiAuthTokenGeneratorController.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    [Route("api/v1/tokens")]
+    [Produces("application/json")]
+    [ApiVersion("1.0")]
+    //TODO: rename class to match the API name
```
Comment (humulla) [inline ApiAuthTokenGenerator/V1/Controllers/ApiAuthTokenGeneratorController.cs:16]: Same here comment can be removed

**Comment by humulla** [comment]
Looks good, tests are implemented nicely, as well as JWT files
Just need some refactored in renaming files to link to API instead of example, and remove comments which have been completed

File: ApiAuthTokenGenerator.Tests/V1/Infrastructure/ExampleContextTests.cs (hunk 3 above)
Comment (LBHMGeorgieva) [reply ApiAuthTokenGenerator.Tests/V1/Infrastructure/ExampleContextTests.cs:9]: Thanks, I will change those once I start working on the DB implementation :)

File: ApiAuthTokenGenerator.Tests/V1/Gateways/ExampleGatewayTests.cs (hunk 2 above)
Comment (LBHMGeorgieva) [reply ApiAuthTokenGenerator.Tests/V1/Gateways/ExampleGatewayTests.cs:8]: Thanks, I will change those once I start working on the gateway implementation

File: ApiAuthTokenGenerator/V1/Controllers/ApiAuthTokenGeneratorController.cs (hunk 5 above)
Comment (LBHMGeorgieva) [reply ApiAuthTokenGenerator/V1/Controllers/ApiAuthTokenGeneratorController.cs:16]: Nice job spotting that, will remove it

**Pull request merged**